src/agent/communicator/include/communicator.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include "defs.hpp"
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+
+namespace beast = boost::beast;
+namespace http = beast::http;
+
+namespace communicator
+{
+    class Communicator
+    {
+    public:
+        Communicator();
+        ~Communicator() {}
+        int SendAuthenticationRequest();
+        int SendRegistrationRequest();
+
+    private:
+        std::string token;
+        http::response<http::dynamic_body> sendHttpRequest(http::verb method,
```

Comment:
What about the `send_http_request` from `src/agent/src/message_task.cpp`?

Reply:
tbh, it would be better to rename the method in the message tasks file, since that function is doing much more than just send a request. It is hidden in an anonymous namespace nevertheless, so there's little risk to have colliding declarations.